src/Exception/UnexpectedResultException.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Search\Exception;
+
+use RuntimeException;
+
+class UnexpectedResultException extends RuntimeException
+{
+    public function __construct(
+        private readonly string $result,
+        string $message = "",
+        int $code = 0,
+        ?\Throwable $previous = null
+    ) {
+        parent::__construct(
+            $message . "\n" . $this->result,
```

Comment:
I'd like advocate against using newlines in exception messages (or any possible log entries in general). Can we use a colon instead?
```suggestion
            $message . ": " . $this->result,
```